File: python/solutions/solution1.py


def multiple_of_three_or_five(x):
    return x % 3 == 0 or x % 5 == 0


def solution(limit):
    return sum(x for x in range(1, limit) if multiple_of_three_or_five(x))


if __name__ == "__main__":
    print(f"The solution is {solution(1000)}")

File: haskell/solutions/solution1.hs
module Solution1 (
    solve1
) where

solve1 :: Int -> Int
solve1 limit = sum([x | x <- [1..(limit-1)], multiple_of_three_or_five x ])

multiple_of_three_or_five :: Int -> Bool
multiple_of_three_or_five x = mod x 3 == 0 || mod x 5 == 0
